app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/ProfileFragmentInActivityTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.github.sdp.tarjetakuna.ui
+
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.sdp.tarjetakuna.MainActivity
+import com.github.sdp.tarjetakuna.R
+import junit.framework.TestCase
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ProfileFragmentInActivityTest {
+
+    private lateinit var activityRule: ActivityScenario<MainActivity>
+
+    @Before
+    fun setUp() {
+        Intents.init()
+        activityRule = ActivityScenario.launch(MainActivity::class.java)
+    }
+
+    @After
+    fun after() {
+        activityRule.close()
+        Intents.release()
+    }
+
+    @Test
+    fun nameEntryChangesSharedPref() {
+        activityRule.onActivity { activity ->
+            activity.changeFragment(R.id.nav_profile)
+        }
+        val newName = "John"
+        // First clear text; okay, Copilote ? ;)
```

Comment:
Here, clearText is useful if you want to run several tests. This avoids errors in the case where the field is not completely empty!

Reply:
Totally! I wrote this comment to prompt Copilote into writing the clearText command (I didn't know its exact syntax and couldn't find it in the autocomp). I ended up looking for it in the doc and forgot to erase my comment when I came back. 🙃